packages/react-devtools-shared/src/devtools/views/Components/EditableKey.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import React, {useRef, useCallback, useEffect, useState} from 'react';
+import styles from './EditableKey.css';
+
+type OverrideKeyFn = (path: Array<string | number>, value: any) => void;
+
+type EditableKeyProps = {|
+  key?: string,
+  overrideKeyFn: OverrideKeyFn,
+|};
+
+export default function EditableKey({
+  key = '',
```

Comment:
`key` and `ref` can't be passed in as props though. (They are extracted and treated specially.) So this will _never_ be anything but undefined (or `""`).

Can we pick a different name for whatever this is? (One that isn't overloaded in React terminology.) I would just suggest naming the state `value` (and `setValue`).

I know this is copying the precedent set by` EditableValue`- but it's borderline mixing controlled and uncontrolled behavior in a way that seems similar similar to [this](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#anti-pattern-erasing-state-when-props-change). Maybe we should name the incoming thing e.g. `initialValue` or something to be more clear?

Reply:
Hm. Did you mean to use this component for existing keys too? Looks like you're only using it for new keys. Just curious!

Reply:
How about calling it `name`, going by the `Object.getOwnPropertyNames` nomenclature?

Reply:
@bvaughn I initially had this idea and played around with it in the `KeyValue` component, but I saw that the PR is getting chunky and decided to not implement it, as it will add more complexity. 
Maybe we can create a new issue for this one and after this get's merged we can start enhancing this feature? What do you think?